frontend/src/modules/Metadata_Forms/components/createMetadataFormModal.js

```diff
@@ -165,19 +165,29 @@ export const CreateMetadataFormModal = (props) => {
               .max(255)
               .required('*Metadata forms name is required'),
             description: Yup.string().max(200),
-            owner: Yup.string().required('*Owner is required'),
-            visibility: Yup.string().required('*Visibility is required'),
+            owner: Yup.string()
+              .oneOf(groups || [])
+              .required('*Owner is required'),
+            visibility: Yup.string()
+              .oneOf(Object.values(visibilityDict))
+              .required('*Visibility is required'),
             environment: Yup.string().when('visibility', {
               is: visibilityDict.Environment,
-              then: Yup.string().required('*Environment is required')
+              then: Yup.string()
+                .oneOf(environmentOptions)
+                .required('*Environment is required')
             }),
             group: Yup.string().when('visibility', {
               is: visibilityDict.Team,
-              then: Yup.string().required('*Team is required')
+              then: Yup.string()
+                .oneOf(groups || [])
+                .required('*Team is required')
             }),
             organization: Yup.string().when('visibility', {
               is: visibilityDict.Organization,
-              then: Yup.string().required('*Organization is required')
+              then: Yup.string()
+                .oneOf(environmentOptions)
```

Comment:
should this be `organizationOptions` not `environmentOptions`

Reply:
[3a598e2](https://github.com/data-dot-all/dataall/pull/1455/commits/3a598e2b1c5f5e3459de6e2c53d0d1e69a047a87)